tools/generation/lib/template.py

```diff
@@ -161,8 +161,9 @@ def _frontmatter(self, case_filename, case_values):
         features += case_values['meta'].get('features', [])
         features += self.attribs['meta'].get('features', [])
         features = list(OrderedDict.fromkeys(features))
```

Comment:
It took me a while to figure out the purpose of this line.
```suggestion
        # remove duplicates while preserving order
        features = list(OrderedDict.fromkeys(features))
```
It's also a good candidate for abstraction à la `indent`:
```python
def without_duplicates(iterable):
    return list(OrderedDict.fromkeys(iterable))
```

Reply:
Yeah, good idea, this makes it less puzzling.